### DISCUSSION TITLE
TS011F_din_smart_relay OTA

### DISCUSSION THREAD

**Discussion by nielsm5** [Koenkk/zigbee2mqtt#12950] [Devices]
Hi,

I recently bought a smart din switch ([TS011F_din_smart_relay](https://www.zigbee2mqtt.io/devices/TS011F_din_smart_relay.html#tuya-ts011f_din_smart_relay)), after installing I immediately noticed the 'automatically turning off' issue as described in https://github.com/Koenkk/zigbee2mqtt/issues/11648. After learning a lot about Zigbee2MQTT I found out my device is running on firmware 1.0.12 (appversion 76). In the beforementioned thread @PietPetrus graciously provided a new version to the OTA repo (see https://github.com/Koenkk/zigbee-OTA/pull/129).

The problem is that even though my device's model and manufacturer info matches with the information provided in the PR, my device is claiming it does not support OTA's.

I've gone through the supported type definitions and there the OTA is configured to use the Zigbee-OTA repo, see: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/tuya.js#L2136

It's my understanding that Zigbee2MQTT uses the [Zigbee-OTA index](https://github.com/Koenkk/zigbee-OTA/blob/master/index.json) file by default as described in: 
https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files

I published a packet to `zigbee2mqtt/bridge/request/device/ota_update/update` hoping I would get more info but alas, it claims it 'does not support OTA updates'.

I had another look at the genOta type and manufacturer ids, see below:
`Read result of 'genOta': {"imageTypeId":65535,"manufacturerId":4417}`
I have no idea if it's related, but it at least does not match the [imageType ](https://github.com/Koenkk/zigbee-OTA/pull/129/files#diff-7aebb122a6ea8a2749d60cb05b7e103c9eae6e2e85e48d2d6cd9e20b63013975R1588) in the Zigbee-OTA index.json file...

Can anyone point me in the right direction?

Regards,
Niels

**Comment by PietPetrus** [comment]
https://github.com/Koenkk/zigbee-herdsman-converters/pull/4353

The bit for OTA has already been merged. If you can wait for a bit longer and update when @Koenkk releases a new version, then OTA will be available.

If you cannot wait than you will have to fiddle a bit with the files to have z2m recognise it as an OTA-capable device and point to the file.

**Reply by nielsm5** [reply]
Sigh, boy do I feel stupid now...
The OTA flag has only recently been added to the supported type definitions.
I figured since there was a previous version it was also OTA capable, never even checked the type definition version.

Thanks

**Reply by PietPetrus** [reply]
Happens to the best of us!
